sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/CookiesShouldBeHttpOnly.cs

```diff
@@ -0,0 +1,124 @@
+
+            field6 = new HttpCookie("c"); // Noncompliant
+
+            Property3.HttpOnly = false; // Noncompliant
+//          ^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+            Property4 = new HttpCookie("c"); // Compliant, HttpOnly is set on the next line
+            Property4.HttpOnly = true;
+
+            // this
+            this.Property5 = new HttpCookie("c"); // Compliant, HttpOnly is set on the next line
+            this.Property5.HttpOnly = true;
+
+            Property6 = new HttpCookie("c"); // Noncompliant
+        }
+    }
+
+    class LocalVariables
+    {
+        private void Cases1()
+        {
+            HttpCookie variable3;
+            var variable1 = new HttpCookie("c"); // Noncompliant
+            var variable2 = new HttpCookie("c") { HttpOnly = false }; // Noncompliant
+            variable3 = new HttpCookie("c"); // Noncompliant
+
+            var variable4 = new HttpCookie("c") { HttpOnly = true }; // Compliant
+            variable4.HttpOnly = false; // Noncompliant
+        }
+
+        private void Cases2()
+        {
+            Class.Method(new HttpCookie("c")); // Noncompliant
+            Class.Method(new HttpCookie("c") { HttpOnly = false }); // Noncompliant
+            Class.Method(new HttpCookie("c") { HttpOnly = true }); // Compliant
+
+            new Class(new HttpCookie("c")); // Noncompliant
+            new Class(new HttpCookie("c") { HttpOnly = false }); // Noncompliant
+        }
+
+        private HttpCookie Return1()
+        {
+            return new HttpCookie("c"); // Noncompliant
+        }
+
+        private HttpCookie Return2()
+        {
+            return new HttpCookie("c") { HttpOnly = true }; // Compliant
+        }
+
+        private void Cases3()
+        {
+            var variable6 = new HttpCookie("c"); // Compliant
+            var variable5 = new HttpCookie("c"); // Noncompliant
+            if (false)
+            {
+                // Do something else
+            }
+            variable6.HttpOnly = true; // This is few statements away from declaration
+
+            HttpCookie variable7;
+            variable7 = new HttpCookie("c"); // Noncompliant, HttpOnly is conditionally set
+            if (true)
+            {
+                variable7.HttpOnly = true;
+            }
+        }
+
+        private HttpCookie Factory1() =>
+            new HttpCookie("c"); // Noncompliant
+
+        private HttpCookie Factory2() =>
+            new HttpCookie("c") { HttpOnly = true }; // Compliant
+    }
+
+    class InvalidSyntax
+    {
+        void Initializers()
+        {
+            var var1 = new HttpCookie( { HttpOnly = true };
+            var var2 = new HttpCookie("c") { HttpOnly }; // Noncompliant
+            var var3 = new HttpCookie("c") { HttpOnly = "asdasd" }; // Noncompliant
+        }
+    }
+
+    class Class
+    {
+        public Class(HttpCookie cookie) { }
+        public static void Method(HttpCookie cookie) { }
+    }
+}
```

Comment:
Other cases:

(A)
1. var myBool = true;
2. var x = new HttpCookie() // not specified
3. x.HttpOnly = myBool;
The constructor doesn't specify, and the mode is set later but to a variable. Is anything reported at lines 2 or 3?

(B): same as (A) except the constructor specifies true.

Reply:
These will raise - we cannot correctly detect the values of variables. Even if we use the symbolic execution engine it will probably be difficult.